pipeflow_lib.tlv

```diff
@@ -381,36 +350,15 @@ m4_popdef(['m4_out_stage'])
 //   |m4_name['']m4_last_phase  (or |m4_out_pipe@(m4_out_stage-1))
 //      @1
 //         $trans_avail   // A transaction is available for consumption.
-m4_define_plus(['m4_self_pipeline'], ['
-m4_pushdef(['m4_top'],         ['$5'])
-m4_pushdef(['m4_name'],        ['$6'])
-m4_pushdef(['m4_in_pipe'],     ['$7'])
-m4_pushdef(['m4_in_stage'],    ['$8'])
-m4_pushdef(['m4_first_phase'], ['$9'])
-m4_pushdef(['m4_last_phase'],  ['$10'])
-m4_pushdef(['m4_out_pipe'],    ['$11'])
-m4_pushdef(['m4_out_stage'],   ['$12'])
-
-'], m4___file__, m4___line__, ['
-m4_forloop(['m4_cycle'], 0, m4_eval((m4_last_phase - m4_first_phase) / 2), ['
-m4_define(['m4_phase'], m4_eval(m4_first_phase + (m4_cycle * 2)))
-m4_define(['m4_in_p'], m4_ifelse(m4_cycle, 0, m4_in_pipe,  m4_name['']m4_decr(m4_phase)))
-m4_define(['m4_in_s'], m4_ifelse(m4_cycle, 0, m4_in_stage, 1))
-m4_define(['m4_out_p'], m4_ifelse(m4_ifelse(m4_out_pipe, , NO_MATCH, )m4_cycle, m4_eval((m4_last_phase - m4_first_phase) / 2), m4_out_pipe, m4_name['']m4_incr(m4_phase)))
-m4_define(['m4_out_s'], m4_ifelse(m4_ifelse(m4_out_stage, , NO_MATCH, )m4_cycle, m4_eval((m4_last_phase - m4_first_phase) / 2), m4_out_stage, 1))
-m4_self_cycle(['$1'], ']']m4___file__['[', ']']m4___line__['[', ['m4_['']self_cycle(...)'], m4_top, m4_in_p, m4_in_s, m4_name['']m4_phase, 1, m4_out_p, m4_out_s)
-'])'],
-
-['
-m4_popdef(['m4_top'])
-m4_popdef(['m4_name'])
-m4_popdef(['m4_in_pipe'])
-m4_popdef(['m4_in_stage'])
-m4_popdef(['m4_first_phase'])
-m4_popdef(['m4_last_phase'])
-m4_popdef(['m4_out_pipe'])
-m4_popdef(['m4_out_stage'])
-'])
+\TLV self_pipeline (/_top,|_name,|_in_pipe,@_in_stage,@_first_phase,@_last_phase,|_out_pipe,@_out_stage)
```

Comment:
@_first_phase and @_last_phase are probably phase numbers, not cycle/stage numbers (phase = cycle * 2, +1 for falling clock edge), so @ can't be used. We should probably change this to use a stage and a boolean arg for falling edge. We can discuss before you implement changes.